compiler/semtempl.nim

```diff
@@ -72,7 +72,7 @@ proc symChoice(c: PContext, n: PNode, s: PSym, r: TSymChoiceRule;
         incl(s.flags, sfUsed)
         markOwnerModuleAsUsed(c, s)
       else:
-        markUsed(c, info, s)
+        markUsed(c, info, s, inGenerics = inGenerics)
```

Comment:
Instead of modifying `markUsed`, just triggering the `if` branch above (i.e. `if isField or inGenerics:` rather than `if isField:`) should be enough.

Also this should apply to templates as well. Another option might be to make this the default behavior for `symChoice` and call `markUsed` manually when the produced node is an `nkSym`. The only place where this isn't already done and isn't in generics/templates seems to be `semexprs.semSymGenericInstantiation`.

Reply:
Shouldn't we check style and suggestSym though?
```nim
  when defined(nimsuggest):
    suggestSym(c.graph, info, s, c.graph.usageSym, false)
  if checkStyle:
    styleCheckUse(c, info, s)
```

Reply:
It would be good to detect it early but then the `when false` case would still be a problem, especially with `--styleCheck:error`. Depends whichever behavior is less annoying.